src/nvim/CMakeLists.txt

```diff
@@ -388,80 +501,246 @@ foreach(hfile ${NVIM_GENERATED_FOR_HEADERS})
+  if(HAVE_WSUGGEST_ATTRIBUTE_PURE)
+    target_compile_options(option_lib INTERFACE -Wsuggest-attribute=pure)
+  endif()
+
+  check_c_compiler_flag(-Wsuggest-attribute=const HAVE_WSUGGEST_ATTRIBUTE_CONST)
+  if(HAVE_WSUGGEST_ATTRIBUTE_CONST)
+    target_compile_options(option_lib INTERFACE -Wsuggest-attribute=const)
+  endif()
+
+  check_c_compiler_flag(-Wsuggest-attribute=malloc HAVE_WSUGGEST_ATTRIBUTE_MALLOC)
+  if(HAVE_WSUGGEST_ATTRIBUTE_MALLOC)
+    target_compile_options(option_lib INTERFACE -Wsuggest-attribute=malloc)
+  endif()
+
+  check_c_compiler_flag(-Wsuggest-attribute=cold HAVE_WSUGGEST_ATTRIBUTE_COLD)
+  if(HAVE_WSUGGEST_ATTRIBUTE_COLD)
+    target_compile_options(option_lib INTERFACE -Wsuggest-attribute=cold)
+  endif()
+endif()
+
+# On FreeBSD 64 math.h uses unguarded C11 extension, which taints clang
+# 3.4.1 used there.
+if(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" AND CMAKE_C_COMPILER_ID MATCHES "Clang")
+  target_compile_options(option_lib INTERFACE -Wno-c11-extensions)
+endif()
+
+# OpenBSD's GCC (4.2.1) doesn't have -Wvla
+check_c_compiler_flag(-Wvla HAS_WVLA_FLAG)
+if(HAS_WVLA_FLAG)
+  target_compile_options(option_lib INTERFACE -Wvla)
+endif()
+
+check_c_compiler_flag(-fno-common HAVE_FNO_COMMON)
+if (HAVE_FNO_COMMON)
+  target_compile_options(option_lib INTERFACE -fno-common)
+endif()
+
+check_c_compiler_flag(-fdiagnostics-color=auto HAS_DIAG_COLOR_FLAG)
+if(HAS_DIAG_COLOR_FLAG)
+  if(CMAKE_GENERATOR MATCHES "Ninja")
+    target_compile_options(option_lib INTERFACE -fdiagnostics-color=always)
+  else()
+    target_compile_options(option_lib INTERFACE -fdiagnostics-color=auto)
+  endif()
+endif()
+
+option(CI_BUILD "CI, extra flags will be set" OFF)
+if(CI_BUILD)
+  message(STATUS "CI build enabled")
+  if(MSVC)
+    target_compile_options(option_lib INTERFACE -WX)
+  else()
+    target_compile_options(option_lib INTERFACE -Werror)
+    if(DEFINED ENV{BUILD_UCHAR})
+      # Get some test coverage for unsigned char
+      target_compile_options(option_lib INTERFACE -funsigned-char)
+    endif()
+  endif()
+endif()
+
+if(UNIX)
+  # -fstack-protector breaks non Unix builds even in Mingw-w64
+  check_c_compiler_flag(-fstack-protector-strong HAS_FSTACK_PROTECTOR_STRONG_FLAG)
+  check_c_compiler_flag(-fstack-protector HAS_FSTACK_PROTECTOR_FLAG)
+
+  if(HAS_FSTACK_PROTECTOR_STRONG_FLAG)
+    target_compile_options(option_lib INTERFACE -fstack-protector-strong)
+    target_link_libraries(link_lib INTERFACE -fstack-protector-strong)
+  elseif(HAS_FSTACK_PROTECTOR_FLAG)
+    target_compile_options(option_lib INTERFACE -fstack-protector --param ssp-buffer-size=4)
+    target_link_libraries(link_lib INTERFACE -fstack-protector --param ssp-buffer-size=4)
   endif()
 endif()
 
-set(NVIM_EXEC_LINK_LIBRARIES ${NVIM_LINK_LIBRARIES} ${LUA_PREFERRED_LIBRARIES})
+add_library(main_lib INTERFACE)
+target_link_libraries(main_lib INTERFACE
+  definition_lib
+  include_lib
+  link_lib
+  option_lib)
```

Comment:
If these are always used together, it can be one single library

Reply:
I was thinking of `texe` on line 938, but I think we should be able to add these libraries as well. On it.